android-components/components/lib/state/src/androidTest/java/mozilla/components/lib/state/ext/ComposeExtensionsKtTest.kt

```diff
@@ -55,6 +55,49 @@ class ComposeExtensionsKtTest {
         }
     }
 
+    @Test
+    fun usingInitialMappedValueWithUpdates() {
+        val loading = "Loading"
+        val content = "Content"
+        val store = Store(
+            initialState = TestState(counter = 0),
+            reducer = ::reducer,
+        )
+
+        val value = mutableListOf<String>()
+
+        rule.setContent {
+            val composeState = store.observeAsComposableState(
+                initialValue = loading,
+                map = { if (it.counter < 5) loading else content },
+            )
+            value.add(composeState.value)
+        }
+        // Initial value when counter is 0.
+        rule.runOnIdle {
+            assertEquals(listOf("Loading"), value)
+        }
+
+        store.dispatchBlockingOnIdle(TestAction.IncrementAction)
+        store.dispatchBlockingOnIdle(TestAction.IncrementAction)
+        store.dispatchBlockingOnIdle(TestAction.IncrementAction)
+        store.dispatchBlockingOnIdle(TestAction.IncrementAction)
+
+        // Value after 4 increments, aka counter is 4. Note that it doesn't recompose here
+        // as the mapped value has stayed the same. We have 1 item in the list and not 5.
```

Comment:
Perhaps move all of these comments before their assertion. This is simply follows the pattern that has already been set by the test case below.